crates/pop-cli/src/commands/build/spec.rs

```diff
@@ -312,162 +475,51 @@ impl BuildSpecCommand {
 			"Need help? Learn more at {}\n",
 			style("https://learn.onpop.io").magenta().underlined()
 		))?;
-
 		Ok("spec")
 	}
-}
 
-/// Guide the user to generate their chain specification.
-async fn guide_user_to_generate_spec(args: BuildSpecCommand) -> anyhow::Result<BuildSpecCommand> {
-	Cli.intro("Generate your chain spec")?;
-
-	// Confirm output path
-	let default_output = format!("./{DEFAULT_SPEC_NAME}");
-	let output_file: String = input("Name of the plain chain spec file. If a path is given, the necessary directories will be created:")
-		.placeholder(&default_output)
-		.default_input(&default_output)
-		.interact()?;
-
-	// Check if specified chain spec already exists, allowing us to default values for prompts
-	let path = Path::new(&output_file);
-	let chain_spec =
-		(path.is_file() && path.exists()).then(|| ChainSpec::from(path).ok()).flatten();
-
-	// Prompt for chain id.
-	let default = chain_spec
-		.as_ref()
-		.and_then(|cs| cs.get_parachain_id())
-		.unwrap_or(DEFAULT_PARA_ID as u64)
-		.to_string();
-	let para_id: u32 = input("What parachain ID should be used?")
-		.placeholder(&default)
-		.default_input(&default)
-		.interact()?;
-
-	// Prompt for chain type.
-	// If relay is Kusama or Polkadot, then Live type is used and user is not prompted.
-	let mut prompt = cliclack::select("Choose the chain type: ".to_string());
-	let default = chain_spec
-		.as_ref()
-		.and_then(|cs| cs.get_chain_type())
-		.and_then(|r| ChainType::from_str(r, true).ok());
-	if let Some(chain_type) = default.as_ref() {
-		prompt = prompt.initial_value(chain_type);
-	}
-	for (i, chain_type) in ChainType::VARIANTS.iter().enumerate() {
-		if default.is_none() && i == 0 {
-			prompt = prompt.initial_value(chain_type);
-		}
-		prompt = prompt.item(
-			chain_type,
-			chain_type.get_message().unwrap_or(chain_type.as_ref()),
-			chain_type.get_detailed_message().unwrap_or_default(),
-		);
+	// Customize a chain specification.
+	fn customize(&self) -> anyhow::Result<()> {
+		let mut chain_spec = ChainSpec::from(&self.output_file)?;
```

Comment:
The integration test (https://github.com/r0gue-io/pop-cli/blob/fix/build-spec/crates/pop-cli/tests/parachain.rs#L13) is failing. After debugging, I identified the issue occurring at this point.
If you want to replicate the test in your chain, the command being executed is:
`pop build spec --output ./target/pop/test-spec.json --id 2222 --type development --chain local --relay paseo-local --genesis-state --genesis-code --protocol-id pop-protocol`

Reply:
This was working on my side now.

Reply:
Is working now. I guess the fix you made here https://github.com/r0gue-io/pop-cli/pull/351/commits/efa35baea4308ecae5d84bc199848ff7f0998f75